fast_llm/engine/multi_stage/stage.py

```diff
@@ -137,7 +137,7 @@ def backward(
         assert self._mode.support_backward
         input_, output = grad_context
         output.backward(output_grad)
-        return input_.grad
+        return input_.grad if input_.grad is not None else torch.zeros_like(input_)
```

Comment:
That's not a good idea, it will cause unnecessary operations when not needed, which is most of the time.